### PR TITLE
Enable goconst, prealloc, stylecheck and dogsled linter

### PR DESCRIPTION
**Pull request by Jaanki**
Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

File: plugin/lighthouse/handler.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ import (
 	"github.com/miekg/dns"
 )
 
+const LightHouse = "lighthouse"
```
Comment (skitt) [inline plugin/lighthouse/handler.go]: Constant names should say what the value does, or is used for, not what it is:

```suggestion
const PluginName = "lighthouse"
```

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/agent/controller/agent.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ import (
 )
 
 const (
-	submarinerIpamGlobalIp = "submariner.io/globalIp"
+	submarinerIpamGlobalIP = "submariner.io/globalIp"
```
Comment (skitt) [inline pkg/agent/controller/agent.go:43]:
```suggestion
	submarinerIPAMGlobalIP = "submariner.io/globalIp"
```

**Review by tpantelis — APPROVED** [review]

**Comment by nyechiel** [comment]
@Jaanki can you please address the review comments? let's get it merged :)

File: pkg/agent/controller/agent.go (hunk 2 above)
Comment (Jaanki) [reply pkg/agent/controller/agent.go:43]: I am changing only those that are flagged by the linter.

Comment (tpantelis) [reply pkg/agent/controller/agent.go:43]: I don't know why the linter didn't flag this - it should have. I think we should change it anyway.

File: plugin/lighthouse/handler.go (hunk 1 above)
Comment (mangelajo) [reply plugin/lighthouse/handler.go]: @Jaanki  ^ Please check Stephen comment. You resolved without addressing.

**Review by mangelajo — CHANGES_REQUESTED** [review]

File: test/e2e/framework/framework.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ import (
 )
 
 const (
-	submarinerIpamGlobalIp = "submariner.io/globalIp"
+	submarinerIpamGlobalIP = "submariner.io/globalIp"
```
Comment (skitt) [inline test/e2e/framework/framework.go:43]:
```suggestion
	submarinerIPAMGlobalIP = "submariner.io/globalIp"
```

same as above.

File: pkg/agent/controller/agent.go (hunk 2 above)
Comment (skitt) [reply pkg/agent/controller/agent.go:43]: The linter has a list of acronyms, which doesn’t include “IPAM”. However while we’re changing names I think we should apply the rule consistently.

**Review by skitt — CHANGES_REQUESTED** [review]

File: pkg/agent/controller/agent.go (hunk 2 above)
Comment (tpantelis) [reply pkg/agent/controller/agent.go:43]: oh I thought you were referring to _IP_ - that should be flagged and I see that @Jaanki did change that. We can change _Ipam_ -> _IPAM_  although in submariner we left it as _Ipam_.

File: .golangci.yml [hunk 4]
```diff
@@ -61,3 +64,9 @@ issues:
     - linters:
         - wsl
       text: "only one cuddle assignment allowed before if statement"
+
+    # Allow dot imports for tests
+    - linters:
+        - stylecheck
+      path: test/*
+      text: "ST1001: should not use dot imports"
```
Comment (tpantelis) [inline .golangci.yml:72]: We should use the same rules as admiral:

 ```
# Allow dot-imports for Gomega BDD directives per idiomatic Gomega
    - linters:
        - stylecheck
        - golint
      text: "should not use dot imports"
      source: ". \"github.com/onsi/gomega\""

    # Allow dot-imports for Ginkgo BDD directives per idiomatic Ginkgo
    - linters:
        - stylecheck
        - golint
      text: "should not use dot imports"
      source: ". \"github.com/onsi/ginkgo\""

```

File: plugin/lighthouse/handler.go (hunk 1 above)
Comment (Jaanki) [reply plugin/lighthouse/handler.go]: I did change it @mangelajo 
https://github.com/submariner-io/lighthouse/pull/466/commits/ec0c600fa3cb5edeb0006285eae378daf7450aa4#diff-0927cc92f1f2487ba548827cdefc8ad44f1e0308b451fe4382ec267d61425d63R29

File: pkg/agent/controller/agent.go (hunk 2 above)
Comment (Jaanki) [reply pkg/agent/controller/agent.go:43]: Yes. In submariner, we left it as it is suggesting that we add a custom linter to check for other such acronyms. Adding a custom linter requires some work which is not a priority IMO. So keeping it unchanged for the sake of consistency.

**Review by dfarrell07 — APPROVED** [review]

**Comment by dfarrell07** [comment]
We need to make sure we squash these. @Jaanki Maybe you can go ahead and `git rebase -i`?

**Review by mangelajo — APPROVED** [review]

**Comment by mangelajo** [comment]
I enabled auto-squash merge, the commit message is handled for now.

**Comment by mangelajo** [comment]
@skitt can you have a look ^ ?

**Review by skitt — APPROVED** [review]

**Pull request merged**